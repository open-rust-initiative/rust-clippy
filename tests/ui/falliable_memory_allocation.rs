#![allow(unused)]
#![warn(clippy::falliable_memory_allocation)]

fn get_untrusted_size() -> usize {
    100
}

fn allocate(_size: usize) {

}

fn main() {
    let size = get_untrusted_size();
    allocate(size);
}
