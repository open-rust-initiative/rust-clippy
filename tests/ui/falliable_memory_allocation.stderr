error: must verify size when allocate memories!
  --> $DIR/falliable_memory_allocation.rs:14:5
   |
LL |     allocate(size);
   |     ^^^^^^^^^^^^^^
   |
note: unverified allocate size used here
  --> $DIR/falliable_memory_allocation.rs:14:14
   |
LL |     allocate(size);
   |              ^^^^

error: must verify null pointer after allocating memories!
  --> $DIR/falliable_memory_allocation.rs:14:5
   |
LL |     allocate(size);
   |     ^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

